api/apps/api/src/modules/clone/import/application/import-archive.handler.spec.ts

```diff
@@ -66,117 +69,54 @@ const getFixtures = async () => {
-        right(
-          Import.fromSnapshot({
-            id: ImportId.create().value,
-            archiveLocation: 'whatever',
-            resourceId: resourceId.value,
-            resourceKind: ResourceKind.Project,
-            importPieces: [
-              {
-                finished: false,
-                order: 0,
-                resourceId: projectId.value,
-                id: `import component unique id`,
-                piece: ClonePiece.ProjectMetadata,
-                uris: [
-                  {
-                    uri: `/tmp/project-metadata-random-uuid.json`,
-                    relativePath: `project-metadata.json`,
-                  },
-                ],
-              },
-              {
-                finished: false,
-                order: 1,
-                resourceId: projectId.value,
-                id: `some other piece`,
-                piece: ClonePiece.PlanningAreaGAdm,
-                uris: [
-                  {
-                    uri: `/tmp/project-planning-area-random-uuid.json`,
-                    relativePath: `planning-area/config.json`,
-                  },
-                ],
-              },
-            ],
-          }),
-        ),
-      );
+      importResourcePieces.mockSequentialPieces();
     },
     GivenExtractingArchiveHasEqualComponents: () => {
-      archiveReader.mock.mockImplementation(async () =>
-        right(
-          Import.fromSnapshot({
-            id: ImportId.create().value,
-            archiveLocation: 'whatever',
-            resourceId: resourceId.value,
-            resourceKind: ResourceKind.Project,
-            importPieces: [
-              {
-                finished: false,
-                order: 2,
-                resourceId: projectId.value,
-                id: `import component unique id`,
-                piece: ClonePiece.ProjectMetadata,
-                uris: [
-                  {
-                    uri: `/tmp/project-metadata-random-uuid.json`,
-                    relativePath: `project-metadata.json`,
-                  },
-                ],
-              },
-              {
-                finished: false,
-                order: 2,
-                resourceId: projectId.value,
-                id: `some other piece`,
-                piece: ClonePiece.PlanningAreaGAdm,
-                uris: [
-                  {
-                    uri: `/tmp/project-planning-area-random-uuid.json`,
-                    relativePath: `planning-area/config.json`,
-                  },
-                ],
-              },
-            ],
-          }),
-        ),
+      importResourcePieces.mockEqualPieces();
+    },
+    WhenRequestingImport: async () => {
+      const importResult = await sut.execute(
+        new ImportArchive(new ArchiveLocation(`whatever`)),
       );
+      if (isRight(importResult))
+        resourceId = new ResourceId(
+          repo.entities[(importResult as Right<string>).right].resourceId,
```

Comment:
As it is already checked that `importResult` is right, we can remove `as Right<...>` 🤔  

```suggestion
          repo.entities[importResult.right].resourceId,
```